botocore/tokens.py

```diff
@@ -324,3 +327,25 @@ def load_token(self):
         return DeferredRefreshableToken(
             self.METHOD, self._refresher, time_fetcher=self._now
         )
+
+    def _resolve_source_profile(self, profiles, initial_profile_config):
+        profile_name = initial_profile_config.get('source_profile', None)
+        source_profile = profiles.get(profile_name, {})
+        visited_profiles = {profile_name}
+        while source_profile.get('source_profile', None):
+            profile_name = source_profile.get('source_profile')
+            source_profile = profiles.get(profile_name, {})
+            if profile_name in visited_profiles:
+                valid = self._has_static_credentials(source_profile)
+                if valid:
+                    break
+                raise InfiniteLoopConfigError(
+                    source_profile=profile_name,
+                    visited_profiles=visited_profiles,
+                )
```

Comment:
Playing around with a bit, it does not seem this error will get called because the credential provider will be resolved before the auth token is resolved. Instead of raising an error here, it probably makes sense to return `None` and delegate any configuration errors related to the assume role credential provider class going forward instead of having it be done in this class which really only cares about `source_profile` so it works with the assume role provider